kanopi/kanopi-pack
# Inline Assets

## Description
> As a developer, I need to provide a way to enable including assets inline.

- This feature added options to specify sets of Inline (`assets.inlineFileTypes`) and Resource  (`assets.resourceFileTypes` - the standard link to separate file) assets.
- By default, all default assets file types are considered resources.

When you set an file types as Inline: 
1. They become Inline
2. If those file types were in Resources, they are removed and the other Default Resources remain with standard processing

When you set both Inline and Resource, only those file types are associated with each Asset type.

When you set only Resources, it overrides the default file types.

## Steps to Validate
1. Select a project with one or more SVG assets referenced in CSS via a `url(/path/to/svg)` link, then pull this update.
2. Set the `[ 'svg' ]` file type to Inline and run either build mode.
3. Confirm the SVG assets are now inserted inline to the CSS instead of linking to the SVG path.
4. Make sure there are assets of another type referenced, for instance, `jpg` and/or `gif`; for the remaining steps, we will assume `jpg`.
5. Set the Resource type to that type, for instance just `[ 'jpg' ]`, restart the build and confirm SVGs are still inlined and JPGs are still loaded (if there were other referenced types now excluded, the process will break, this is expected)
6. If your process did not break yet, change the Resource file list to something else like `[ 'doc' ]` and rebuild
7. The process should now break on all `jpg` files
8. Remove the two settings and confirm both SVG and JPG are now included as Resource assets
